lib/mix/lib/mix/tasks/test.ex

```diff
@@ -229,9 +229,18 @@ defmodule Mix.Tasks.Test do
   end
 
   defp parse_files([], test_paths) do
+    warn_about_incorrect_extensions_in(test_paths)
+
     test_paths
   end
 
+  defp warn_about_incorrect_extensions_in(test_paths) do
+    for file <- Mix.Utils.extract_files(test_paths, "*_test.ex") do
+      IO.puts "Warning: #{file} has incorrect extension and won't be run. " <>
```

Comment:
`Mix.shell` can be configured by the application, it's not limited to IO.  

Elixir ships with two `Mix.Shell` implementations, IO and Process, which can be found [here](https://github.com/elixir-lang/elixir/tree/4f68c4f10502e0f54a21093bb9a33957e63a9ac4/lib/mix/lib/mix/shell).